src/zzz_od/api/utils/validation.py

```
@@ -0,0 +1,242 @@
+    @staticmethod
+    def safe_read_yaml(file_path: str) -> Dict[str, Any]:
+        """安全读取YAML文件"""
+        try:
+            FileOperationHelper.check_file_exists(file_path, "配置")
+            FileOperationHelper.check_file_permissions(file_path, "读取")
+
+            with open(file_path, 'r', encoding='utf-8') as f:
+                content = yaml.safe_load(f)
+                return content if content is not None else {}
+        except (yaml.YAMLError, UnicodeDecodeError) as e:
+            raise ConfigurationError(f"配置文件格式错误: {file_path}, 错误: {str(e)}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)